protobuf/module_msg.proto

```diff
@@ -6,385 +6,869 @@ package bess.pb;
+   */
   message Field {
-    uint64 size = 1;
-    uint64 mask = 4;
+    uint64 size = 1; /// The length, in bytes, of the field to inspect.
+    uint64 mask = 4; /// A bitmask over the field to specify which bits to inspect (default 0xff).
     oneof position {
-      string name = 2;
-      int64 offset = 3;
+      string name = 2; ///Metadata attribute name, if field resides in metadata.
+      int64 offset = 3; /// The offset, in bytes, from the start of the packet that the field resides in (if field resides in packet data)..
     }
   }
-  repeated Field fields = 1;
+  repeated Field fields = 1; ///A list of ExactMatch Fields
 }
 
+/**
+ * The FlowGen module generates simulated TCP flows of packets with correct SYN/FIN flags and sequence numbers.
+ * This module is useful for testing, e.g., a NAT module or other flow-aware code.
+ * Packets are generated off a base, "template" packet by modifying the IP src/dst and TCP src/dst. By default, only the ports are changed and will be modified by incrementing the template ports by up to 2000 more than the template values.
+ *
+ * __Input Gates__: 0
+ * __Output Gates__: 1
+ */
 message FlowGenArg {
-  bytes template = 1;
-  double pps = 2;
-  double flow_rate = 3;
-  double flow_duration = 4;
-  string arrival = 5;
-  string duration = 6;
-  bool quick_rampup = 7;
-  uint32 ip_src_range = 8;
-  uint32 ip_dst_range = 9;
-  uint32 port_src_range = 10;
-  uint32 port_dst_range = 11;
-}
-
+  bytes template = 1; /// The packet "template". All data packets are derived from this template and contain the same payload.
+  double pps = 2; /// The total number of pps to generate.
+  double flow_rate = 3; /// The number of new flows to create every second. flow_rate must be <= pps.
+  double flow_duration = 4; /// The lifetime of a flow in seconds.
+  string arrival = 5; /// The packet arrival distribution -- must be either "uniform" or "exponential"
+  string duration = 6; /// The flow duration distribution -- must be either "uniform" or "pareto"
+  bool quick_rampup = 7; /// Whether or not to populate the flowgenerator with initial flows (start generating full pps rate immediately) or to wait for new flows to be generated naturall (all flows have a SYN packet).
+  uint32 ip_src_range = 8; /// When generating new flows, FlowGen modifies the template packet by changing the IP src, incrementing it by at most ip_src_range.
+  uint32 ip_dst_range = 9; /// When generating new flows, FlowGen modifies the template packet by changing the IP dst, incrementing it by at most ip_dst_range.
+  uint32 port_src_range = 10; /// When generating new flows, FlowGen modifies the template packet by changing the TCP port, incrementing it by at most port_src_range.
+  uint32 port_dst_range = 11; /// When generating new flows, FlowGen modifies the template packet by changing the TCP dst port, incrementing it by at most port_dst_range.
```

Comment:
For these ranges, what does it mean for something to be "incremented" at most by a range?